aws/resource_aws_ec2_tag.go

```diff
@@ -103,20 +117,30 @@ func resourceAwsEc2TagRead(d *schema.ResourceData, meta interface{}) error {
 			},
 		})
 
-		// tag not found _yet_
-		if len(tags.Tags) == 0 {
-			return resource.RetryableError(fmt.Errorf("tag not found"))
-		}
-
 		if err != nil {
 			return resource.RetryableError(err)
 		}
 
+		// tag not found _yet_
+		if len(tags.Tags) == 0 {
+			return resource.RetryableError(NewResourceNotExists("tag not found"))
+		}
+
 		return nil
 	})
 
 	if retryError != nil {
-		return fmt.Errorf("[ERROR] Tag %s not found on resource %s", key, id)
+		if _, ok := retryError.(*ResourceNotExists); !ok {
+			return fmt.Errorf("[ERROR] Could not read tag %s on resource %s", key, id)
+		}
+	}
```

Comment:
`resource.Retry` doesn't exit until the timeout has occurred. It will check for `resource.RetryableError` values before returning `retryError` here.

Reply:
Yes, I think that is my intention too. I added the custom error to differentiate between resource does not exist after timeout vs the api call is failing for some other reason.

Reply:
This is kind of implied by the retry failing to find the tag. I'm going to merge this for now, but I've not seen this pattern before and it might end up needing to be refactored in order to be accepted up stream.

Reply:
There could be failures in making the actual API call. That would not mean that resource does not exist. We do not know the actual resource state. However, if we get a response without failure and there are no tags, then we know that it actually does not exist.